### PR TITLE
Added Pod Annotation support to the PodConfigurationSpec

### PR DESCRIPTION
**Pull request by kalinon**
Pod annotations are used for various things. At Diamanti we use them for QOS and scheduling. By allowing them to be set for the job pods, we can benchmark different configurations.

### CODE REVIEW

File: api/v1alpha1/k8s_types.go [hunk 1]
```diff
@@ -106,6 +105,14 @@ func (v *VolumeSpec) Validate() (ok bool, err error) {
 
 // PodConfigurationSpec contains the configuration for the benchmark pods
 type PodConfigurationSpec struct {
+
+	// Annotations is an unstructured key value map stored with a resource that may be
```
Comment (julio-santiesteban) [inline api/v1alpha1/k8s_types.go]: The comment should start with the element name: "PodAnnotations is an..." Reference: https://blog.golang.org/godoc-documenting-go-code

File: controllers/iperf3/serverdeployment.go [hunk 2]
```diff
@@ -73,8 +73,9 @@ func NewServerDeployment(cr *perfv1alpha1.Iperf3) *appsv1.Deployment {
 
 	deployment := appsv1.Deployment{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      serverDeploymentName(cr),
-			Namespace: cr.Namespace,
+			Name:        serverDeploymentName(cr),
+			Namespace:   cr.Namespace,
+			Annotations: cr.Spec.ServerConfiguration.PodConfigurationSpec.PodAnnotations,
```
Comment (julio-santiesteban) [inline controllers/iperf3/serverdeployment.go]: In this case the name `PodAnnotations` is quite misleading. You're also using the map to annotate the deployment, which I think is fine, but then it should be named simply `Annotations`.

Comment (kalinon) [reply controllers/iperf3/serverdeployment.go]: I agree with this. i was following the formatting of the other sections in the `PodConfigurationSpec` such as `PodLabels` or `PodScheduling`.

Comment (kalinon) [reply controllers/iperf3/serverdeployment.go]: @julio-santiesteban can you confirm you would like me to change the naming?

Comment (julio-santiesteban) [reply controllers/iperf3/serverdeployment.go]: `PodLabels` and `PodScheduling` only applies to the pod. In contrast, you apply the given annotations to the deployment / job, too, so yes, I do think simply `Annotations` would be a better name.

Comment (kalinon) [reply controllers/iperf3/serverdeployment.go]: @julio-santiesteban i have updated the branch and changed the value to Annotations

**Pull request merged**